src/utils.rs

```diff
@@ -0,0 +1,286 @@
+use std::time::SystemTime;
+
+use x509_cert::{certificate::CertificateInner, crl::CertificateList};
+
+/// A module for serializing and deserializing certificate chains.
+pub mod cert_chain {
+    use serde::{Deserialize, de, ser};
+    use x509_cert::{certificate::CertificateInner, der::EncodePem};
+
+    use super::cert_chain_processor;
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Vec<CertificateInner>, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let s = <String>::deserialize(deserializer)?;
+        let pem = Box::new(s.as_bytes());
+        cert_chain_processor::load_pem_chain_bpf_friendly(&pem).map_err(de::Error::custom)
+    }
+
+    pub fn serialize<S>(certs: &Vec<CertificateInner>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut string = String::new();
+        for cert in certs {
+            string.push_str(
+                &cert
+                    .to_pem(p256::pkcs8::LineEnding::LF)
+                    .map_err(ser::Error::custom)?,
+            );
+        }
+        serializer.serialize_str(&string)
+    }
+}
+
+/// A module for serializing and deserializing CRLs.
+pub mod crl_optional {
```

Comment:
I believe this module is no longer needed.

Reply:
Aah yes ! Let me clean it up !